### ISSUE TITLE
Add subjects that actions can be performed on

### ISSUE THREAD

**Issue by philippgille** [adamabdelhamed/PowerArgs#124]
As I understand the documentation, with PowerArgs you can currently implement these two kinds of CLI:
- `myApp -some "arg"`
- `myApp someAction -some "arg"`

But some CLIs have another "level", which is *subjects*. For example the Docker CLI:
- `docker container ls -a`
- `docker container run --name some-redis -d redis`

Where `docker` is the application, `container` the subject and `ls` / `run` etc. the action, with arguments only applying to the specific action.

Is this possible with PowerArgs? If not, I think this might be a powerful addition for complex CLIs.

**Comment by adamabdelhamed** [comment]
Yes you can.  See the calculator example on the main readme of the project.

**Comment by philippgille** [comment]
The usage example for the calculator example is `Calculator.exe add -Value1 1 -Value2 5`, which is `app action -param`, not `app subject action -param`. Or did I misunderstand the example?
